freqtrade/optimize/backtesting.py

```diff
@@ -358,6 +358,94 @@ def _get_close_rate(self, sell_row: Tuple, trade: LocalTrade, sell: SellCheckTup
         Get close rate for backtesting result
         """
         # Special handling if high or low hit STOP_LOSS or ROI
+        is_short = trade.is_short or False
+        if is_short:
+            return self._get_short_close_rate(sell_row, trade, sell, trade_dur)
+        else:
+            return self._get_long_close_rate(sell_row, trade, sell, trade_dur)
+
+    def _get_short_close_rate(self, sell_row: Tuple, trade: LocalTrade, sell: SellCheckTuple,
+                              trade_dur: int) -> float:
+        leverage = trade.leverage or 1.0
+        if sell.sell_type in (SellType.STOP_LOSS, SellType.TRAILING_STOP_LOSS):
+            if trade.stop_loss < sell_row[LOW_IDX]:
+                # our stoploss was already lower than candle high,
+                # possibly due to a cancelled trade exit.
+                # sell at open price.
+                return sell_row[OPEN_IDX]
+
+            # Special case: trailing triggers within same candle as trade opened. Assume most
+            # pessimistic price movement, which is moving just enough to arm stoploss and
+            # immediately going down to stop price.
+            if sell.sell_type == SellType.TRAILING_STOP_LOSS and trade_dur == 0:
+                if (
+                    not self.strategy.use_custom_stoploss and self.strategy.trailing_stop
+                    and self.strategy.trailing_only_offset_is_reached
+                    and self.strategy.trailing_stop_positive_offset is not None
+                    and self.strategy.trailing_stop_positive
+                ):
+                    # Worst case: price reaches stop_positive_offset and dives down.
+                    stop_rate = (sell_row[OPEN_IDX] *
+                                 (1 - abs(self.strategy.trailing_stop_positive_offset) +
+                                  abs(self.strategy.trailing_stop_positive)))
+                else:
+                    # Worst case: price ticks tiny bit above open and dives down.
+                    stop_rate = sell_row[OPEN_IDX] * (1 + abs(trade.stop_loss_pct / leverage))
+                    assert stop_rate > sell_row[HIGH_IDX]
```

Comment:
This will have to use `LOW_IDX` - otherwise we can ONLY get prices outside of the candle.